sphinxcontrib/redoc.py

```diff
@@ -30,25 +32,40 @@ def render(app):
         # relies on them.
         ctx.setdefault('opts', {})
 
-        # The 'spec' may contain either HTTP(s) link or filesystem path. In
-        # case of later we need to copy the spec into output directory, as
-        # otherwise it won't be available when the result is deployed.
-        if not ctx['spec'].startswith(('http', 'https')):
-            specpath = os.path.join(app.builder.outdir, '_specs')
-            specname = os.path.basename(ctx['spec'])
-
-            ensuredir(specpath)
-
-            copyfile(
-                # Since the path may be relative it should be joined with
-                # base URI which is a path of directory with conf.py in
-                # our case.
-                os.path.join(app.confdir, ctx['spec']),
-                os.path.join(specpath, specname))
-
-            # The link inside rendered document must refer to a new location,
-            # the place where it has being copied to.
-            ctx['spec'] = os.path.join('_specs', specname)
+        # "Offline" mode: the JSON spec is embedded in the page
+        # TODO: only works for local files: add remote files download ?
+        if ctx.get('offline') is True:
+            # Parse & dump the spec to have it as properly formatted json
+            with io.open(os.path.join(app.confdir, ctx['spec'])) as specfp:
+                # Dumb heuristic
+                parser_module = json if ctx['spec'].endswith("json") else yaml
```

Comment:
YAML could be used to parse JSON. For the sake of simplicity, I'd go with YAML loading.

P.S: I mean, `json` is usually much faster than `YAML` in parsing, but this is not performance critical part. I mean end users wouldn't even notice that.

Reply:
I didn't even know this was possible. Simpler, OK with me.